coral-hive/src/main/java/com/linkedin/coral/hive/hive2rel/parsetree/ParseTreeBuilder.java

```diff
@@ -222,38 +222,55 @@ private SqlNode visitLateralViewInternal(ASTNode node, ParseContext ctx, boolean
 
   private SqlNode visitLateralViewExplode(List<SqlNode> sqlNodes, List<SqlNode> aliasOperands,
       SqlCall tableFunctionCall, boolean isOuter) {
-    checkState(aliasOperands.size() == 3);
+    final int operandCount = aliasOperands.size();
+    // array if operandCount == 3, map if operandCount == 4
```

Comment:
suggestion: would be nice to add example syntax for array and map, to be clear why operandCount == 3 for array and 4 for map. Something like:

`map if operandCount == 4:  LATERAL VIEW EXPLODE(op0) op1 AS op2, op3 `

Reply:
Thanks for suggestion, added.